### ISSUE TITLE
[Feature] Extend .ga workflow format with parameters value

### ISSUE THREAD

**Issue by jibe-b** [galaxyproject/galaxy#4181]
The .ga format contains the steps, input and outputs.

Using the Bioblend API, it is possible by executing `run_workflow` to have a workflow executed with a specific set of parameters. [source](https://bioblend.readthedocs.io/en/latest/_modules/bioblend/galaxy/workflows.html#WorkflowClient.run_workflow)

Exporting parameters of, say, last execution of a worklow on the platform to a .ga file would enable to keep track of parameters.

It would also enable to write steps to execute and parameters within a single file instead of two ones.

**Comment by bgruening** [comment]
@jibe-b the workflow spec supports the parameters. Try to create a workflow and set parameters in the UI, export it and have a look at the json file.
If you execute a workflow you can (at least via the UI) export the history to a new workflow, which then should have all parameters included as well.

**Comment by mandiayba** [comment]
@bgruening are the values of the parameters saved into the workflow (.ga) ? I think what @jibe-b is asking is to fill a workflow (.ga) (including parameters and values for the tools) to the a api function and let it execute it without additional setting.

**Comment by shiltemann** [comment]
yes these are saved as well, see the `tool_state` attribute of the tools in the .ga file ([example](https://github.com/galaxyproject/training-material/blob/master/topics/metagenomics/tutorials/mothur-miseq-sop/workflows/mothur-miseq-sop.ga#L390))
